chaos_genius/core/utils/kpi_validation.py

```diff
@@ -1,59 +1,72 @@
 """Provides utility functions for validating KPIs."""
 
+from datetime import datetime
 import logging
-from typing import Any, Dict, List, Tuple, Union
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 import pandas as pd
 from pandas.api.types import is_datetime64_any_dtype as is_datetime
 
 from chaos_genius.core.rca.root_cause_analysis import SUPPORTED_AGGREGATIONS
 from chaos_genius.core.utils.data_loader import DataLoader
+from chaos_genius.databases.models.data_source_model import DataSource
 from chaos_genius.settings import MAX_ROWS_FOR_DEEPDRILLS
 
 KPI_VALIDATION_TAIL_SIZE = 1000
 
 logger = logging.getLogger(__name__)
 
 
-def validate_kpi(kpi_info: Dict[str, Any], data_source: Dict[str, Any]) -> Tuple[bool, str]:
+def validate_kpi(kpi_info: Dict[str, Any], check_tz_aware: bool = False) -> Tuple[bool, str, Optional[bool]]:
     """Load data for KPI and invoke all validation checks.
 
     :param kpi_info: Dictionary with all params for the KPI
     :type kpi_info: Dict[str, Any]
-    :param data_source: Dictionary describing the data source
-    :type data_source: Dict[str, Any]
-    :return: Returns a tuple with the status as a bool and a status message
-    :rtype: Tuple[bool, str]
+    :param check_tz_aware: Bool for checking if the data is timezone aware
+    :return: Returns a tuple with the status as a bool, a status message and None if check_tz_aware is False otherwise a bool telling whether the data is timezone aware
+    :rtype: Tuple[bool, str, Optional[bool]]
     """
     try:
         df = DataLoader(
             kpi_info, tail=KPI_VALIDATION_TAIL_SIZE, validation=True
         ).get_data()
         logger.info(f"Created df with {len(df)} rows for validation")
-    except Exception as e:  # noqa: B902
+    except Exception as e:
         logger.error("Unable to load data for KPI validation", exc_info=1)
-        return False, "Could not load data. Error: " + str(e)
+        return False, f"Could not load data. Error: {str(e)}"
```

Comment:
This returns a tuple of 2 values. The caller is expecting 3 values and tries to assign to 3 variables. This will cause an error.

Reply:
(the type checker also checks this)

Reply:
Fixed